### DISCUSSION TITLE
Plot IPF with hexagonal symmetry

### DISCUSSION THREAD

**Discussion by darrencpagan** [neperfepx/neper#391] [Q&A]
Hi,

I'm currently simulating an HCP polycrystal. When I generate the tesselation, I indicate hexagonal symmetry.

However, when plotting the orientations using the visualization module, how do I ensure the ipf is plotted using the hexagonal standard triangle? Is this known automatically from the data or is there a secondary option?

Also the manual -dataeltcolscheme section indicates the the ipf coloring syntax as `ipf[(<dir>)]` when I believe the square brackets are not necessary.

Thank you,
Darren

**Comment by rquey** [comment]
Hi Darren,

Cubic crystal symmetry is always assumed (see [-datacellcolscheme](https://neper.info/doc/neper_v.html#cmdoption-datacellcolscheme)). Neper would indeed determine the crystal symmetry from the tessellation file if it was able to do something else than cubic... but only cubic is implemented.

`ipf[(<dir>)]` is actually the correct syntax following our [writing conventions](https://neper.info/doc/introduction.html#conventions-used-in-this-manual)... It means that the argument can be `ipf` or `ipf(<dir>)`, where `ipf(<dir>)` can be `ipf(x)`, `ipf(y)` or `ipf(z)`... The square brackets (`[ ]`) denote something optional, `<` and `>` surround a metasyntaxtic variable, and the parentheses are just parentheses and do not mean anything special.

**Comment by darrencpagan** [comment]
Got it, thank you. Will hexagonal symmetry for IPF plotting be implemented in the future?

**Reply by rquey** [reply]
In the future yes, but as I'm not working on hexagonal myself, it will only happen when I need it or take the time to generalize this... or when someone wants to contribute it.
